src/components/userComponents/Footer/Footer.tsx

```diff
@@ -0,0 +1,557 @@
+'use client';
+
+import React, { useState } from 'react';
+import { BiErrorCircle } from 'react-icons/bi';
+import { IoIosCheckmarkCircleOutline } from 'react-icons/io';
+import Link from 'next/link';
+import supabase from '../../../supabase/client';
+import { useWebDeviceDetection } from '../../../context/WindowWidthContext/WindowWidthContext';
+
+type EmailInputProps = {
+  inputValueName: string;
+  inputValueEmail: string;
+  subscribed: boolean;
+  handleNameChange: (e: any) => void;
+  handleEmailChange: (e: any) => void;
+  handleSubmit: (e: any) => void;
+  showError: boolean;
+  errorMsg: string;
+};
+
+/**
+ *
+ * @param root0 overall
+ * @param root0.inputValueName name input
+ * @param root0.inputValueEmail email input
+ * @param root0.handleSubmit submit change
+ * @param root0.showError shows error
+ * @param root0.errorMsg error message
+ * @param root0.handleNameChange when change name input
+ * @param root0.handleEmailChange when change email input
+ * @param root0.subscribed if you successfully subscribed
+ * @returns email input component
+ */
+function MobileInput({
+  inputValueName,
+  inputValueEmail,
+  subscribed,
+  handleNameChange,
+  handleEmailChange,
+  handleSubmit,
+  showError,
+  errorMsg,
+}: EmailInputProps) {
+  return (
+    <div className="flex flex-col justify-center">
+      <div className="bg-hunterGreen mt-2 pb-16">
+        {subscribed && (
```

Comment:
@jxmoose @sarahhpeng  what is intended way to check if someone is subscribed?  Currently if I subscribe, change pages, then come back to original page, I am asked to subscribe again. Could this cause issues in the backend?

Efficient possible solution:  use a react context to track if a user has submitted an email in current instance.  Can check this to conditionally render this component without having to requery supabase.

Reply:
stored in sessionstorage but could also store in localstorage. sessionstorage persists only across while have site open while localstorage basically stays forever even if close site lol which one is better?

Reply:
@jxmoose i think localstorage might be better. and also change that column in the emails table in supabase so that the emails are unique.